src/components/Product/Filters.js

```diff
@@ -0,0 +1,44 @@
+const getSortedData = (state, data) => {
+  switch (state.sortBy) {
+    case "LOW_TO_HIGH":
+      return [...data].sort((a, b) => {
+        return a.price - b.price;
+      });
+
+    case "HIGH_TO_LOW":
+      return [...data].sort((a, b) => {
+        return b.price - a.price;
+      });
+
+    default:
+      return data;
+  }
+};
+
+const getFilteredData = (data, includeOutOfStock, fastDelivery) => {
+  var filteredData = [...data];
+  if (includeOutOfStock && !fastDelivery) {
+    filteredData = [...filteredData].filter((items) => items.inStock);
+  } else if (fastDelivery && !includeOutOfStock) {
+    filteredData = [...filteredData].filter((items) => items.fastDelivery);
+  } else if (fastDelivery && includeOutOfStock) {
+    filteredData = [...filteredData]
```

Comment:
`filter` returns a new array, you don't need to spread it, it won't mutate the original array.

Reply:
okay i will replace it, thanks for pointing out